impeller/typographer/text_frame.cc

```diff
@@ -46,4 +46,36 @@ bool TextFrame::HasColor() const {
   return has_color_;
 }
 
+bool TextFrame::MaybeHasOverlapping() const {
+  if (runs_.size() > 1) {
+    return true;
+  }
+  auto glyph_positions = runs_[0].GetGlyphPositions();
+  if (glyph_positions.size() > 10) {
```

Comment:
Why this?

What happens if there are 100 `TextFrame` objects each with 10 glyphs? This seems like the wrong level for optimization.

Reply:
If there is more than 1 run or more than 10 glyphs, we assume that its not worth computing the overlap - instead just assume the text frame has overlapping contents and don't apply the optimization. Seems reasonable to me?

Reply:
L51, we don't

Reply:
So we can't get multiple `TextFrame`s in a single pass?

Reply:
Sorry, I'm not sure what you mean?

Reply:
discussed offline, we need to do these checks because text rendering behaves differently than solid color fills. The optimization has a limitation in the number of entities we check per frame which is now documented on the PaintPassDelegate